lib/ansible/modules/crypto/luks_device.py

```diff
@@ -0,0 +1,457 @@
+    module_args = dict(
+        state=dict(type='str', choices=['present', 'absent'], required=False, default='present'),
+        device=dict(type='str', required=False),
+        open=dict(type='bool', required=False, default=False),
+        name=dict(type='str', required=False),
+        key=dict(type='path', required=False),
+        new_key=dict(type='path', required=False),
+        remove_key=dict(type='path', required=False)
+    )
+
+    # seed the result dict in the object
+    result = dict(
+        changed=False,
+        name=None
+    )
+
+    module = AnsibleModule(argument_spec=module_args,
+                           supports_check_mode=False)
+
+    crypt = CryptHandler(module)
+    conditions = ConditionsHandler(module)
+
+    # The conditions are in order to allow more operations in one run.
+    # (e.g. create luks and add a key to it)
+
+    # luks create
+    if conditions.luks_create():
+        try:
+            crypt.run_luks_create(module.params['device'],
+                                  module.params['key'])
+        except ValueError as e:
+            module.fail_json(msg="luks_device error: %s" % e)
+        result['changed'] = True
+
+    # luks open
+    if conditions.luks_open():
+        name = module.params['name']
+        if name is None:
+            try:
+                name = crypt.generate_luks_name(module.params['device'])
+            except ValueError as e:
+                module.fail_json(msg="luks_device error: %s" % e)
+        try:
+            crypt.run_luks_open(module.params['device'],
+                                module.params['key'],
+                                name)
+        except ValueError as e:
+            module.fail_json(msg="luks_device error: %s" % e)
+        result['name'] = name
+        result['changed'] = True
+
+    # luks close
+    if conditions.luks_close():
+        if module.params['device'] is not None:
+            name = crypt.get_container_name_by_device(module.params['device'])
+        else:
+            name = module.params['name']
+        crypt.run_luks_close(name)
+        result['changed'] = True
+
+    # luks add key
+    if conditions.luks_add_key():
+        try:
+            crypt.run_luks_add_key(module.params['device'],
+                                   module.params['key'],
+                                   module.params['new_key'])
+        except ValueError as e:
+            module.fail_json(msg="luks_device error: %s" % e)
+        result['changed'] = True
+
+    # luks remove key
+    if conditions.luks_remove_key():
+        try:
+            crypt.run_luks_remove_key(module.params['device'],
+                                      module.params['remove_key'])
+        except ValueError as e:
+            module.fail_json(msg="luks_device error: %s" % e)
+        result['changed'] = True
+
+    # luks remove
+    if conditions.luks_remove():
+        crypt.run_luks_remove(module.params['device'])
```

Comment:
There's another `try/except` missing here.

Reply:
`conditions.luks_remove` does not raise any exception to be caught.

Reply:
`run_luks_remove()` does raise `ValueError`.

Reply:
Here's another line which can raise `ValueError` (call to `run_luks_remove()`).